wire/wire.go

```diff
@@ -13,22 +13,60 @@
 // limitations under the License.
 
 // Package wire contains directives for Wire code generation.
+// For an overview of working with Wire, see the user guide at
+// https://github.com/google/go-cloud/blob/master/wire/README.md
+//
+// The directives in this package are used as input to the Wire code
+// generation tool. The entry point of Wire's analysis are injector functions:
+// function templates denoted by only containing a call to Build. Build names
+// a set of providers and the Wire code generation tool builds a directed
+// acylic graph of the providers' output types. The generated generated code
+// will fill in the function template by calling the providers from the
```

Comment:
How about instead of "calling", something like "... by using the providers in the provider set to instantiate any needed types."?

Not all providers are "called", e.g. struct providers, and I think it might add clarity to be explicit about why it's calling the providers.

Reply:
Done.